lifting_veil/agents/dqn_agent_new.py

```diff
@@ -311,11 +315,24 @@ def _train_step(self):
                 else:
                     loss_weights = jnp.ones(self.replay_elements['state'].shape[0])
 
-                self.optimizer, loss, mean_loss = train(
-                    self.network_def, self.target_network_params, self.optimizer, self.replay_elements['state'],
-                    self.replay_elements['action'], self.replay_elements['next_state'], self.replay_elements['reward'],
-                    self.replay_elements['terminal'], loss_weights, self.cumulative_gamma, self._target_opt,
-                    self._mse_inf, self._tau, self._alpha, self._clip_value_min, self._rng)
+                self.optimizer_state, self.online_params, loss = train(
+                    self.network_def,
+                    self.online_params,
+                    self.target_network_params,
+                    self.optimizer_state, 
```

Comment:
you still need to pass `self.optimizer` as the 4th parameter

Reply:
Yes, perfect!